### ISSUE TITLE
[Package Request] socat 1.8.0.1

### ISSUE THREAD

**Issue by mietzen** [SynoCommunity/spksrc#6258]
### Software Name

socat

### Brief Description

Socat 1.8.0.0 has been released with SynoCli Network Tools 2.4-18. Unfortunately 1.8.0.0 brings some unexpected behavior changes, that are allegedly fixed in 1.8.0.1 released on 2024-08-24:

> Failures report for 1.8.0.0
> 
> A group of failures in Socat version 1.8.0.0 have been reported for commands that worked fine in previous versions (regression). They affect addresses using IP protocols without explicit IP version, with options bind and/or range. Examples are TCP-LISTEN, UDP, UDP-RECV, OPENSSL. The error messages are:
> 
>     E syntax error in "localhost"
>     E syntax error in range "127.0.0.1/8" of unspecified address family: use <addr>:<mask>
>     E xioopen_ipdgram_listen(): unknown address family 0
>     E getaddrinfo("127.0.0.1", "NULL", {0x01,10,2,17}, {}): Address family for hostname not supported
>     E xioparsenetwork_ip6("localhost:255.255.255.255",,): missing mask bits delimiter '/'
>     E missing brackets for IPv6 range definition "127.0.0.1/8"
> 
> These issues will be addressed by version 1.8.0.1, planned for August; possible workarounds are:
> 
>     Use Socat up to version 1.7.4.4 (or 1.7.4.5 from git)
>     Apply option -4 (does not help in some cases)
>     Use explicit IP version in address key word like TCP4-LISTEN, UDP4 etc;
>     Add address option pf=ip4 (esp.with OPENSSL)

http://www.dest-unreach.org/socat/

### Website

http://www.dest-unreach.org/socat/

### Build/Installation documentation

http://www.dest-unreach.org/socat/

### Source code

http://www.dest-unreach.org/socat/

### OpenSource License

GPLv2

**Comment by chickenandpork** [comment]
Hey @mietzen are you able to share the actual commands?  I would not be the one to likely fix, but I know it’s easier to test whether it’s resolved if there are test cases.

thanks!

**Comment by mietzen** [comment]
Sure:

```
$ socat TCP-LISTEN:5002,fork,range=192.168.10.0/24 TCP:10.10.0.50:5001
2024/10/07 06:25:42 socat[9637] E syntax error in range "192.168.10.0/24" of unspecified address family: use <addr>:<mask>
```

**Comment by mietzen** [comment]
> This is an issue of socat 1.8.0.0 fixed in 1.8.0.1 see http://www.dest-unreach.org/socat/

Hence the Request to release the newer version 😉
